src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -921,19 +922,49 @@ public override AwaitExpressionInfo GetAwaitExpressionInfo(AwaitExpressionSyntax
             }
 
             var bound = GetLowerBoundNode(node);
-            BoundAwaitableInfo awaitableInfo = (((bound as BoundExpressionStatement)?.Expression ?? bound) as BoundAwaitExpression)?.AwaitableInfo;
+            BoundAwaitableInfo? awaitableInfo = (((bound as BoundExpressionStatement)?.Expression ?? bound) as BoundAwaitExpression)?.AwaitableInfo;
+            return GetAwaitExpressionInfo(awaitableInfo);
+        }
+
+        public override AwaitExpressionInfo GetAwaitExpressionInfo(LocalDeclarationStatementSyntax node)
+        {
+            if (node.UsingKeyword.IsKind(SyntaxKind.None) || node.AwaitKeyword.IsKind(SyntaxKind.None))
+            {
+                throw new ArgumentException("node is not an await using declaration", nameof(node));
```

Comment:
>"node is not an await using declaration" [](http://example.com/codeflow?start=44&length=40)

Probably should be localized
I realize we're not very consistent on this, but it is best practice
[https://learn.microsoft.com/en-us/dotnet/standard/exceptions/best-practices-for-exceptions#include-a-localized-string-message](https://learn.microsoft.com/en-us/dotnet/standard/exceptions/best-practices-for-exceptions#include-a-localized-string-message)